File: 96-二叉树的中序遍历.cpp
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {

        stack<TreeNode*> mystack;
        vector<int> myvector;

        TreeNode* current = root;

        while(current || !mystack.empty()){

            while(current){
                mystack.push(current);
                current = current->left;
            }
            
            current = mystack.top();
            mystack.pop();
            myvector.push_back(current->val);
            current = current->right;

        }

        return myvector;
    }
};

File: 96-二叉树的中序遍历.py
class Solution:
    
    def inorderTraversal(self, root: TreeNode) -> List[int]:

        result = []

        def inorder(root):
            if root:
                inorder(root.left)
                result.append(root.val)
                inorder(root.right)
        
        inorder(root)

        return result
